src/librados/librados.cc

```diff
@@ -3279,6 +3303,27 @@ extern "C" int rados_write_full(rados_ioctx_t io, const char *o, const char *buf
   return retval;
 }
 
+extern "C" int rados_writesame(rados_ioctx_t io,
+				const char *o,
+				const char *buf,
+				size_t data_len,
+				size_t write_len,
+				uint64_t off)
+{
+  tracepoint(librados, rados_writesame_enter, io, o, buf, data_len, write_len, off);
+  if ((data_len > UINT_MAX/2) || (write_len > UINT_MAX/2))
+    return -E2BIG;
+  if ((data_len == 0) || (write_len % data_len))
+    return -EINVAL;
```

Comment:
You're right, I'll fix this (and rados_aio_writesame) in the next round - thanks!